src/RelayClient.ts

```diff
@@ -254,55 +255,36 @@ class RelayClient extends EnvelopingEventEmitter {
 
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-ignore FIXME: remove the ts ignore once the method is consumed
-  private _prepareRelayHttpRequest = async (
-    { feesReceiver }: HubInfo,
-    { request, relayData }: RelayRequest
+  private _prepareHttpRequest = async (
+    relayRequest: RelayRequest
   ): Promise<EnvelopingTxRequest> => {
-    const { callForwarder } = relayData;
-    const { relayHub: relayHubAddress } = request;
-
-    if (!callForwarder) {
-      throw new Error('Call forwarder must be defined.');
-    }
-
-    if (
-      !feesReceiver ||
-      feesReceiver === constants.AddressZero ||
-      !isAddress(feesReceiver)
-    ) {
-      throw new Error('FeesReceiver has to be a valid non-zero address');
-    }
-
-    const updatedData: EnvelopingRequestData = {
-      ...relayData,
-      feesReceiver,
-    };
-
+    const {
+      request: { relayHub },
+      relayData: { callForwarder },
+    } = relayRequest;
     const chainId = (await this._provider.getNetwork()).chainId;
     const accountManager = new AccountManager(this._provider, chainId);
-
     const relayMaxNonce =
       (await this._provider.getTransactionCount(callForwarder)) +
       this._envelopingConfig.maxRelayNonceGap;
-
     const metadata: EnvelopingMetadata = {
-      signature: await accountManager.sign({
-        request,
-        relayData: updatedData,
-      }),
-      relayHubAddress,
+      relayHubAddress: await relayHub,
+      signature: await accountManager.sign(relayRequest),
       relayMaxNonce,
     };
+    const httpRequest: EnvelopingTxRequest = {
+      relayRequest,
+      metadata,
+    };
 
     this.emit(envelopingEvents['sign-request']);
+    log.info(
+      `Created HTTP ${
+        isDeployRequest(relayRequest) ? 'deploy' : 'relay'
```

Comment:
I think that is longer necessary to have the `isSmartWalletDeploy`, I think that using the `isDeployRequest` is more practical